versions.tf

```diff
@@ -0,0 +1,57 @@
+armonik_versions = {
+  infra     = "2.10.1"
+  core      = "0.8.3"
+  api       = "0.3.1"
+  gui       = "0.7.2"
+  extcsharp = "0.8.1"
+  samples   = "2.10.0"
+}
+armonik_images = {
+  infra = [
+  ]
+  core = [
+    "dockerhubaneo/armonik_pollingagent",
+    "dockerhubaneo/armonik_control_metrics",
+    "dockerhubaneo/armonik_control_partition_metrics",
+    "dockerhubaneo/armonik_control",
+    "dockerhubaneo/armonik_core_stream_test_worker",
+    "dockerhubaneo/armonik_core_stream_test_client",
+    "dockerhubaneo/armonik_core_htcmock_test_worker",
+    "dockerhubaneo/armonik_core_htcmock_test_client",
+    "dockerhubaneo/armonik_core_bench_test_worker",
+    "dockerhubaneo/armonik_core_bench_test_client",
+  ]
+  api = [
+  ]
+  gui = [
+    "dockerhubaneo/armonik_admin_app",
+    "dockerhubaneo/armonik_admin_api",
+  ]
+  extcsharp = [
+    "dockerhubaneo/armonik_worker_dll",
+  ]
+  samples = [
+  ]
+}
+image_tags = {
+  "k8s.gcr.io/autoscaling/cluster-autoscaler"                      = "v1.23.0"
+  "k8s.gcr.io/metrics-server/metrics-server"                       = "v0.6.1"
+  "ghcr.io/kedacore/keda"                                          = "2.8.0"
+  "ghcr.io/kedacore/keda-metrics-apiserver"                        = "2.8.0"
+  "public.ecr.aws/aws-ec2/aws-node-termination-handler"            = "v1.15.0"
+  "amazon/aws-efs-csi-driver"                                      = "v1.4.3"
+  "public.ecr.aws/eks-distro/kubernetes-csi/livenessprobe"         = "v2.2.0-eks-1-18-13"
+  "public.ecr.aws/eks-distro/kubernetes-csi/node-driver-registrar" = "v2.1.0-eks-1-18-13"
+  "public.ecr.aws/eks-distro/kubernetes-csi/external-provisioner"  = "v2.1.1-eks-1-18-13"
+  "symptoma/activemq"                                              = "5.16.4"
+  "mongo"                                                          = "5.0.9"
+  "redis"                                                          = "6.2.7"
+  "datalust/seq"                                                   = "2022.1"
+  "grafana/grafana"                                                = "9.2.1"
+  "prom/node-exporter"                                             = "v1.3.1"
+  "prom/prometheus"                                                = "v2.36.1"
+  "fluent/fluent-bit"                                              = "1.9.9"
+  "rtsp/mongosh"                                                   = "1.5.4"
+  "nginx"                                                          = "1.23.2"
+  "nginxinc/nginx-unprivileged"                                    = "1.23.2"
+}
```

Comment:
Preferably have a consistent and uniform nomenclature for all images. Use for example a map "key:value" where:
* key is the image registry
* value is the image's tag

I prefer using a JSON file instead of TF file.

Reply:
About the JSON file, I would prefer keep the `.tfvars` instead but it is more readable, but I don't really mind either way.

About the "key:value", the goal of this structure is to give the default tag for each image, so it is really a dictionary and not a mere list. We need the lookup property of the map. Moreover, the default tag must be unique for a given image, and a list cannot provide this guarantee.

Reply:
Semi-structured JSON files are more common for any user/developer/devops.

Use dictionary and not a list, example for armonik control plane:
{
   "armonik-controle-plane": {
      "registry": "dockerhubaneo/armonik_control",
      "tag": "0.8.3",
   }
}

Reply:
> Use dictionary and not a list, example for armonik control plane:
> {
> "armonik-controle-plane": {
> "registry": "dockerhubaneo/armonik_control",
> "tag": "0.8.3",
> }
> }

This is exactly the design I wanted to avoid with this PR. I want people who deploy their infra to be able to change the images in whatever way they want. I just give an image name as default value of the parameter, and give the default tag for whatever image they need if no tag is provided. The module just says: "if you want `dockerhubaneo/armonik_control`, here is the default tag for it.

This is especially important for workers as we can have multiple workers with different images in a single deployment.

Reply:
I agree with @lemaitre-aneo as we do not want to have multiple places where the same tag appears. It will simplify the updates to new versions.